src/ui/src/components/blueprints/BlueprintsAutogen.vue

```diff
@@ -59,43 +59,52 @@ onMounted(() => {
  * The generation happens with simple ids (aig1, aig2, ...).
  * Component ids are altered to preserve uniqueness across blueprints.
  */
-function alterIds(components: Component[]) {
+function alterIds(actions: AutogenAction[]) {
 	const mapping: Record<string, string> = {};
 
-	// Switch ids
-
-	components.forEach((c) => {
-		const newId = generateNewComponentId();
-		mapping[c.id] = newId;
-		c.id = newId;
-	});
+	actions.forEach((a) => {
```

Comment:
I think one letter variable is OK for a single line function, but here it's almost 40 lines. You should use a more explicit name to improve readabilite

```suggestion
	actions.forEach((action) => {
```